File: Practica_1/src/gestorAplicacion/Producto.java

package gestorAplicacion;
public class Producto {
    private int fechaVencimiento;
    private String nombre;
    private int precioCompra;
    private int precioVenta;
    private long codigo;
    private int fechaIngreso;
    private int fechaVenta;
    
    //constructor
    
    public Producto(int fechaVencimiento, String nombre, int precioCompra, int precioVenta, long codigo, int fechaIngreso) {
        this.fechaVencimiento = fechaVencimiento;
        this.nombre = nombre;
        this.precioCompra = precioCompra;
        this.precioVenta = precioVenta;
        this.codigo = codigo;
        this.fechaIngreso = fechaIngreso;
        
        Informe.getInventario().add(this);
        
    }
    
         //getters y setters

    public int getFechaVencimiento() {
        return fechaVencimiento;
    }

    public void setFechaVencimiento(int fechaVencimiento) {
        this.fechaVencimiento = fechaVencimiento;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getPrecioCompra() {
        return precioCompra;
    }

    public void setPrecioCompra(int precioCompra) {
        this.precioCompra = precioCompra;
    }

    public int getPrecioVenta() {
        return precioVenta;
    }

    public void setPrecioVenta(int precioVenta) {
        this.precioVenta = precioVenta;
    }

    public long getCodigo() {
        return codigo;
    }

    public void setCodigo(long codigo) {
        this.codigo = codigo;
    }

    public int getFechaIngreso() {
        return fechaIngreso;
    }

    public void setFechaIngreso(int fechaIngreso) {
        this.fechaIngreso = fechaIngreso;
    }

    public int getFechaVenta() {
        return fechaVenta;
    }

    public void setFechaVenta(int fechaVenta) {
        this.fechaVenta = fechaVenta;
    }
    
}

File: Practica_2/src/gestorAplicacion/gestion/Producto.py
class Producto:

    _productos = []
    _numeroProducto = 0

def __init__(self,trabajador = None, cliente = None, estado ="", fecchaVenta = None, tipo="" ):
    self._trabajador = trabajador
    self._cliente = cliente
    self._estado = estado 
    self._nombre = nombre 
    self._precioVenta = precioVenta
    self._codigo = codigo 
    self._fechaVenta = fechaVenta
    self._tipo = tipo

    Producto._productos.append(self)

    
def getTrabajador(self):
    return self._trabajador

def setTrabajador(self, trabajador):
    self._trabajador = trabajador

def getCliente(self):
    return self._cliente 

def setCliente(self, cliente):
    self._cliente = cliente 

def getEstado(self):
    return self._estado 

def setEstado(self, estado):
    self._estado = estado

def getNombre(self):
    return self._nombre

def setNombre(self, nombre):
    self._nombre = nombre 

def getPrecioVenta(self):
    return self._precioVenta

def setPrecioVenta(self, precioVenta):
    self._precioVenta = precioVenta 

def getCodigo(Self):
    return self._codigo

def setCodigo(self, codigo):
    self._codigo = codigo 

def getFechaVenta(self):
    return self._fechaVenta 

def setFechaVenta(self, fechaVenta):
    self._fechaVenta = fechaVenta 

def getTipo(self):
    return self._tipo

def setTipo(self, tipo):
    self._tipo = tipo

@classmethod
def getProductos(cls):
    return cls._productos

@classmethod
def setProductos(cls, productos):
    cls.productos = productos

@classmethod
    def getNumeroProductos(cls):
        return cls._numeroProducto

@classmethod
    def setNumeroPedido(cls, numeroProducto):
        cls._numeroProducto = numeroProducto



    
